src/components/FeatureViews/AdminPortal/PanelManager.js

```diff
@@ -148,15 +102,65 @@ function PanelManager(props) {
                     fields[field][key] = defaults[key];
                 }
             })
-            console.log(fields)
-            return fields
         })
-    }
+        return fields
+    };
+
+    useEffect(() => {
+        if(records.length !== recordList.length) {
+            let parsedRecordList = records.map((record) => ({
+                ...record,
+                editing: false,
+            }));
+            setRecordList(parsedRecordList);
+        }
+    }, [records]);
     const fieldsWithDefaults = addDefaults(props.fields);
+
+    useEffect(() => {
+        const constructDirectoryObject = (fieldInfo, recordArray) => {
```

Comment:
Can you put some documentation of why you decided to construct the directory object this way?